### PR TITLE
Tn patterns

### PR DESCRIPTION
**Pull request by thomasgnuttall**
Beginning of pattern extraction code. And a waveform player that visualises waveform, allows playback and overlays annotations

### CODE REVIEW

File: compiam/melody/pattern/sancara_search/extraction/pitch.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    :param p: Pitch value in Hz
+    :type p: float
+    :param tonic: Tonic value in Hz
+    :type tonic: float
+
+    :return: Pitch value, <p> in cents above <tonic>
+    :rtype: float
+    """
+    return 1200*math.log(p/tonic, 2) if p else None
```
Comment (genisplaja) [inline compiam/melody/pattern/sancara_search/extraction/pitch.py:29]: same here: todo esto podría estar en `compiam/utils/pitch.py`. We can leave that for this other PR though.

File: compiam/visualisation/waveform_player/waveform-playlist/annotations.html [hunk 2]
```diff
@@ -0,0 +1,231 @@
+<!DOCTYPE html>
```
Comment (genisplaja) [inline compiam/visualisation/waveform_player/waveform-playlist/annotations.html:1]: very cool stuff here :)

File: notebooks/melody/pattern_search.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+audio_path = "/Volumes/Shruti/asplab2/cae-invar/audio/lara_wim/spleeter/2018_11_13_am_Sec_4_P1_Atana_V.2.mp3"
```
Comment (genisplaja) [inline notebooks/melody/pattern_search.py:1]: Is this file needed?

File: requirements.txt [hunk 4]
```diff
@@ -8,6 +9,8 @@ pytsmod==0.3.3
 scikit-learn~=0.24.2
 tqdm~=4.64.1
 mirdata==0.3.6
+#skimage
```
Comment (genisplaja) [inline requirements.txt:12]: Do we want this one or not?

**Review by genisplaja — APPROVED** [review]
Just some minor comments and discussions but looks very nice!

File: compiam/melody/pattern/sancara_search/extraction/io.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        return ParamList(value)
+
+    yaml.add_constructor('!paramlist', constructor_paramlist)
+    yaml.add_constructor('!dottedname', constructor_dottedname)
+
+    if not os.path.isfile(path):
+        return None
+    with open(path) as f:
+        d = yaml.load(f, Loader=yaml.FullLoader)   
+    return d
```
[the remark this answers is not in the document]
Comment (thomasgnuttall) [reply compiam/melody/pattern/sancara_search/extraction/io.py:154]: Yep, this is temporary actually I do plan to move this out

File: notebooks/melody/pattern_search.py (hunk 3 above)
Comment (thomasgnuttall) [reply notebooks/melody/pattern_search.py:1]: no
